### ISSUE TITLE
[BUG] "Path is too long" error occur if reopen local onedrive database

### ISSUE THREAD

**Issue by scruel** [PhilippC/keepass2android#2354]
### Checks

- [X] I have read the FAQ section, searched the open issues, and still think this is a new bug.

### Describe the bug you encountered:

- Install Onedrive
- Open File
- System File Selector
- select OneDrive/database.kdbx file and unlock the database
- restart the app and try to reopen the database

(I don't want to use the OAuth way)

```
Path is too long. Path /data/data/keepass2android.keepass2android/no_backup/contentcom.microsoft.skydrive.content.StorageAccessProviderdocumentcontent%3A%2F%2Fcom.microsoft.skydrive.content.metadata...
```

### What version of Keepass2Android are you using?

1.09e-r7

### Which version of Android are you on?

Android 13, Android 12

**Comment by jiucenglou** [comment]
I encountered same error here

**Comment by milloenchen** [comment]
Same here. Is there a workaround?

**Comment by PhilippC** [comment]
I cannot reproduce this. When you are on the password screen or in the list of recent files, how does the filename look like? For me it's something like "content://database.kdbx". Is it longer for you?

**Comment by PhilippC** [comment]
hmm, for me that looks very different... In order to open files from Onedrive, I first had to install the OneDrive app (https://play.google.com/store/apps/details?id=com.microsoft.skydrive&hl=de&gl=US). Is this what you have as well, or are you using something else?

**Comment by milloenchen** [comment]
Yes, I have the same app installed. Is it because the kdbx file is not in OneDrive's root folder? It's in like the third? level folder: Files > Documents > keepass.

**Comment by milloenchen** [comment]
@PhilippC the issue is solved it seems. I disabled the automatic battery optimisation in Android for the OneDrive app and now it works! So it was probably an issue with the app authorisation / battery optimisation.
Thank you. 😊
